File: Modulo06/Heranca/Professor.cpp
#include "Professor.h"
#include <iostream>
using namespace std;

Professor::Professor(string nome, int CPF, double salario) : 
Pessoa(nome, CPF), salario(salario) {
}

double Professor::getSalario() const {
    return salario;
}

void Professor::imprime() const {
    Pessoa::imprime();
    cout << "Salário: " << this->getSalario() << endl;
}

File: Modulo12/Professor-Java/br/ufscar/dc/poo/Professor.java
package br.ufscar.dc.poo;

public class Professor extends Pessoa {

    private double salario;

    public Professor(String nome, int CPF, double salario) {
        super(nome, CPF);
        this.salario = salario;
    }

    public double getSalario() {
        return salario;
    }

    public void imprime() {
        super.imprime();
        System.out.println("Salário: " + this.getSalario());
    }
}